source/frontend/src/layouts/PublicLayout.tsx

```diff
@@ -1,19 +1,148 @@
-import React from 'react';
+import axios, { AxiosError } from 'axios';
+import React, { useCallback, useEffect, useState } from 'react';
 import { ErrorBoundary } from 'react-error-boundary';
 import LoadingBar from 'react-redux-loading-bar';
 
 import ErrorModal from '@/components/common/ErrorModal';
 import { Footer, Header } from '@/components/layout';
+import HealthcheckView, {
+  IHealthCheckIssue,
+} from '@/components/layout/Healthcheck/HealthcheckView';
+import ISystemCheck from '@/hooks/pims-api/interfaces/ISystemCheck';
+import { useApiHealth } from '@/hooks/pims-api/useApiHealth';
+import useKeycloakWrapper from '@/hooks/useKeycloakWrapper';
+import { IApiError } from '@/interfaces/IApiError';
+import { useTenant } from '@/tenants/useTenant';
 
 import FooterStyled from './Footer';
 import HeaderStyled from './Header';
+import HealthCheckStyled from './Healthcheck';
+import { HealthcheckMessagesTypesEnum } from './models/HealthcheckMessagesTypes';
 import * as Styled from './styles';
 
 const PublicLayout: React.FC<React.PropsWithChildren<React.HTMLAttributes<HTMLElement>>> = ({
   children,
   ...rest
 }) => {
-  return (
+  const [systemChecked, setSystemChecked] = useState<boolean>(null);
```

Comment:
I don't think this logic should be added to the publicLayout. The PublicLayout should be stateless, and just accept params for these health check items.